source/common/grpc/google_grpc_utils.cc

```diff
@@ -78,7 +78,9 @@ Buffer::InstancePtr GoogleGrpcUtils::makeBufferInstance(const grpc::ByteBuffer&
   // NB: ByteBuffer::Dump moves the data out of the ByteBuffer so we need to ensure that the
   // lifetime of the Slice(s) exceeds our Buffer::Instance.
   std::vector<grpc::Slice> slices;
-  byte_buffer.Dump(&slices);
+  if (!byte_buffer.Dump(&slices).ok()) {
```

Comment:
Is there a way to mock this for testing?

Reply:
Even the gRPC library provided by Google doesn't test for Dump failure.  Here is the only test:

TEST_F(ByteBufferTest, Dump) {                                                                                                                                                                                                                                                                                                                                                                                                            
  grpc_slice hello = grpc_slice_from_copied_string(kContent1);                                                                                                                                                                                                                                                                                                                                                                            
  grpc_slice world = grpc_slice_from_copied_string(kContent2);                                                                                                                                                                                                                                                                                                                                                                            
  std::vector<Slice> slices;                                                                                                                                                                                                                                                                                                                                                                                                              
  slices.push_back(Slice(hello, Slice::STEAL_REF));                                                                                                                                                                                                                                                                                                                                                                                       
  slices.push_back(Slice(world, Slice::STEAL_REF));                                                                                                                                                                                                                                                                                                                                                                                       
  ByteBuffer buffer(&slices[0], 2);                                                                                                                                                                                                                                                                                                                                                                                                       
  slices.clear();                                                                                                                                                                                                                                                                                                                                                                                                                         
  (void)buffer.Dump(&slices);                                                                                                                                                                                                                                                                                                                                                                                                             
  EXPECT_TRUE(SliceEqual(slices[0], hello));                                                                                                                                                                                                                                                                                                                                                                                              
  EXPECT_TRUE(SliceEqual(slices[1], world));                                                                                                                                                                                                                                                                                                                                                                                              
}

Reply:
Yeah, this is fine. I think if this was Envoy code we would want a test, but it's not worth the effort to deal with this in external deps that aren't written with mocking in mind.